Objects/unicodeobject.c

```diff
@@ -6086,17 +6082,13 @@ _PyUnicode_DecodeUnicodeEscape(const char *s,
 
       error:
         endinpos = s-starts;
-        writer.min_length = end - s + writer.pos;
         if (unicode_decode_call_errorhandler_writer(
                 errors, &errorHandler,
                 "unicodeescape", message,
                 &starts, &end, &startinpos, &endinpos, &exc, &s,
                 &writer)) {
             goto onError;
         }
-        if (_PyUnicodeWriter_Prepare(&writer, writer.min_length, 127) < 0) {
-            goto onError;
```

Comment:
You are right. The right assert is `assert(end - s <= writer.size - writer.pos)`. Seems `_PyUnicodeWriter_Prepare()` is incorrectly used here and in `unicode_decode_call_errorhandler_writer()`. And `min_length` may be inconsistently used in different decoders.